cpp/lib/ClientConfigParser.cpp

```diff
@@ -0,0 +1,315 @@
+/**
+ * Copyright (c) 2024 Snowflake Computing
+ */
+
+#include "snowflake/ClientConfigParser.hpp"
+#include "snowflake/Exceptions.hpp"
+#include "../logger/SFLogger.hpp"
+#include "client_config_parser.h"
+#include "memory.h"
+
+#include <fstream>
+#include <sstream>
+#include <set>
+
+#undef snprintf
+#include <boost/filesystem.hpp>
+
+#ifndef _WIN32 
+#include <dlfcn.h>
+#endif
+
+using namespace Snowflake::Client;
+
+namespace
+{
+  // constants
+  const std::string SF_CLIENT_CONFIG_FILE_NAME("sf_client_config.json");
+  const std::string SF_CLIENT_CONFIG_ENV_NAME("SF_CLIENT_CONFIG_FILE");
+  std::set<std::string> KnownCommonEntries = {"log_level", "log_path"};
+}
+
+////////////////////////////////////////////////////////////////////////////////
+void load_client_config(
```

Comment:
When we wrapping c++ into c interface better to catch any possible exception since c interface doesn't expect exception.